<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Usando as Exceções</title>
<link rel="stylesheet" type="text/css" href="dbstyle.css">
<meta name="generator" content="DocBook XSL Stylesheets V1.79.2">
<link rel="home" href="index.html" title="Guia de Referência do Programador">
<link rel="up" href="zend.exception.html" title="Zend_Exception">
<link rel="prev" href="zend.exception.html" title="Zend_Exception">
<link rel="next" href="zend.exception.basic.html" title="Uso básico">
</head>
<body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF">
<div class="navheader"><table width="100%" summary="Navigation header">
<tr><th colspan="3" align="center">Usando as Exceções</th></tr>
<tr>
<td width="20%" align="left">
<a accesskey="p" href="zend.exception.html">Anterior</a> </td>
<th width="60%" align="center">Zend_Exception</th>
<td width="20%" align="right"> <a accesskey="n" href="zend.exception.basic.html">Próxima</a>
</td>
</tr>
</table></div>
<div class="sect1">
<div class="titlepage"><div><div><h2 class="title" style="clear: both">
<a name="zend.exception.using"></a>Usando as Exceções</h2></div></div></div>
    

    <p>
        <code class="classname">Zend_Exception</code> é simplesmente a classe base para
        todas as exceções lançadas dentro do Zend Framework.
    </p>

    <div class="example">
<a name="zend.exception.using.example"></a><p class="title"><b>Exemplo 388. Tratando uma Exceção</b></p>
<div class="example-contents">
        

        <p>
            O código a seguir demonstra como tratar uma exceção lançada em uma
            classe do Zend Framework:
        </p>

        <pre class="programlisting">
try {
    // Chamar Zend_Loader::loadClass() com uma classe inexistente irá causar
    // uma exceção para ser lançada em Zend_Loader:
    Zend_Loader::loadClass('nonexistantclass');
} catch (Zend_Exception $e) {
    echo "Tratando a exceção: " . get_class($e) . "\n";
    echo "Mensagem: " . $e-&gt;getMessage() . "\n";
    // Outro código para recuperar do erro
}
</pre>
    </div>
</div>
<br class="example-break">

    <p>
        <code class="classname">Zend_Exception</code> pode ser usado como uma classe de exceção pega-tudo
        em um bloco de tratamento para capturar todas as exceções lançadas pelas classes do
        Zend Framework. Isso pode ser útil quando o programa não consegue se recuperar
        através do tratamento de um tipo específico de exceção.
    </p>

    <p>
        A documentação de cada componente e classe do Zend Framework irá conter informações
        específicas sobre os métodos para lançar exceções, as circunstâncias que fazem com
        que uma exceção seja lançada, e os diversos tipos de exceção que podem ser lançadas.
    </p>
</div>
<div class="navfooter"><table width="100%" summary="Navigation footer">
<tr>
<td width="40%" align="left">
<a accesskey="p" href="zend.exception.html">Anterior</a> </td>
<td width="20%" align="center"><a accesskey="u" href="zend.exception.html">Acima</a></td>
<td width="40%" align="right"> <a accesskey="n" href="zend.exception.basic.html">Próxima</a>
</td>
</tr>
<tr>
<td width="40%" align="left" valign="top">Zend_Exception </td>
<td width="20%" align="center"><a accesskey="h" href="index.html">Principal</a></td>
<td width="40%" align="right" valign="top"> Uso básico</td>
</tr>
</table></div>
<div class="revinfo"></div>
</body>
</html>
